apps/condo/domains/organization/schema/ReplaceOrganizationEmployeeRoleService.js

```diff
@@ -0,0 +1,222 @@
+        variable: ['data'],
+        code: BAD_USER_INPUT,
+        type: 'ROLES_MUST_BE_DIFFERENT',
+        message: 'The old role and new role must be different',
+        messageForUser: 'api.organization.ReplaceOrganizationEmployeeRole.ROLES_MUST_BE_DIFFERENT',
+    },
+    ROLES_ARE_BEING_PROCESSED: {
+        mutation: 'replaceOrganizationEmployeeRole',
+        variable: ['data'],
+        code: TOO_MANY_REQUESTS,
+        type: 'ROLES_ARE_BEING_PROCESSED',
+        message: 'These roles are already being processed. Please try again a little later',
+        messageForUser: 'api.organization.ReplaceOrganizationEmployeeRole.ROLES_ARE_BEING_PROCESSED',
+    },
+}
+
+const ReplaceOrganizationEmployeeRoleService = new GQLCustomSchema('ReplaceOrganizationEmployeeRoleService', {
+    types: [
+        {
+            access: true,
+            type: 'input ReplaceOrganizationEmployeeRoleInput { dv: Int!, sender: SenderFieldInput!, organization: OrganizationWhereUniqueInput!, oldRole: OrganizationEmployeeRoleWhereUniqueInput!, newRole: OrganizationEmployeeRoleWhereUniqueInput!, withDeletionOldRole: Boolean! }',
+        },
+        {
+            access: true,
+            type: 'type ReplaceOrganizationEmployeeRoleOutput { status: String! }',
+        },
+    ],
+
+    mutations: [
+        {
+            access: access.canReplaceOrganizationEmployeeRole,
+            schema: 'replaceOrganizationEmployeeRole(data: ReplaceOrganizationEmployeeRoleInput!): ReplaceOrganizationEmployeeRoleOutput',
+            doc: {
+                summary: 'Replaces old role "A" with new role "B" for all employees with role "A"',
+                description: 'Replaces old role "A" with new role "B" for all employees with role "A". ' +
+                    'By default, old role is retained. If you pass the “withDeletionOldRole” flag, then old role will be deleted.',
+                errors: ERRORS,
+            },
+            resolver: async (parent, args, context) => {
+                const { data } = args
+
+                const sender = get(data, 'sender', null)
+                const dv = get(data, 'dv', null)
+                const organizationId = get(data, 'organization.id', null)
+                const oldRoleId = get(data, 'oldRole.id', null)
+                const newRoleId = get(data, 'newRole.id', null)
+                const withDeletionOldRole = get(data, 'withDeletionOldRole', false)
+
+                checkDvAndSender(data, ERRORS.DV_VERSION_MISMATCH, ERRORS.WRONG_SENDER_FORMAT, context)
+
+                const organization = await getById('Organization', organizationId)
+                if (!organization || organization.deletedAt) throw new GQLError(ERRORS.ORGANIZATION_NOT_FOUND, context)
+
+                const oldRole = await getById('OrganizationEmployeeRole', oldRoleId)
+                if (!oldRole || oldRole.deletedAt || oldRole.organization !== organizationId) throw new GQLError(ERRORS.OLD_ROLE_NOT_FOUND, context)
```

Comment:
Maybe `getByConditions('', { deletedAt, organization: {id: ...} })`?

Reply:
Why do you want to change `getById` to `getByConditions`?
This will not change anything significantly

Only in `getById` it is additionally checked that the ID is not empty